### ISSUE TITLE
renderdoccmd capture

### ISSUE THREAD

**Issue by z00318884** [baldurk/renderdoc#1355]
## Description

when use "renderdoccmd capture" to capture my application on linux，It also enters the if branch，not into else branch，it‘s OK？
Also "renderdoc__replay__marker" is in "renderdoccmd" not in "librenderdoc.so", Why  "dlsym(RTLD_DEFAULT, identifier)" can find this symbol？

void library_loaded()
{
  if(LibraryHooks::Detect("renderdoc__replay__marker"))
  {
    RDCDEBUG("Not creating hooks - in replay app");

    RenderDoc::Inst().SetReplayApp(true);

    RenderDoc::Inst().Initialise();

    return;
  }
  else
  {
    RenderDoc::Inst().Initialise();

    const char *capturefile = Process::GetEnvVariable("RENDERDOC_CAPFILE");
    const char *opts = Process::GetEnvVariable("RENDERDOC_CAPOPTS");
........
}

## Environment

* RenderDoc build: 2019-04-13 nightly build.
* Operating System:  Ubuntu Linux,
* API: . OpenGL,

thank you!

**Comment by baldurk** [comment]
I'm not sure I entirely understand your problem. Can you clarify, do you mean that your application is being identified as a replay application and so it's not capturing? If that's the case then something must be exporting the `renderdoc__replay__marker` marker symbol in your application's process, you'd need to use a debugger or similar to determine why it's being exported.

**Comment by z00318884** [comment]
It’s not a problem without capturing, but the code also enters the if branch when it is captured, which makes me confused.

for example： 
~/mywork$ gdb --args ./renderdoccmd capture -d -c ./renderdoccmd replay X111.rdc
GNU gdb (Ubuntu 7.7.1-0ubuntu5~14.04.2) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./renderdoccmd...done.
(gdb) b posix_libentry.cpp:34
No source file named posix_libentry.cpp.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (posix_libentry.cpp:34) pending.
(gdb) r
Starting program: /home/mywork/renderdoc/build/bin/renderdoccmd capture -d -c ./renderdoccmd replay X111.rdc
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Breakpoint 1, library_loaded () at /home/mywork/renderdoc/renderdoc/os/posix/posix_libentry.cpp:34
34        if(LibraryHooks::Detect("renderdoc__replay__marker"))
(gdb) n
36          RDCDEBUG("Not creating hooks - in replay app");//////**why enter this branch?**??
(gdb)

**Comment by baldurk** [comment]
You're capturing `renderdoccmd` itself which is as you've already mentioned marked as a replay application. That's why it's identified as such when capturing, and I'm really not sure what else would be expected to happen.

This is not supported, RenderDoc is intended for capturing your own application - not itself.

**Comment by z00318884** [comment]
This is just an example of mine. I found that all the programs are like this. There is actually no real load program, so it should have nothing to do with what program to capture.
Even if you just write something unknow, it can go to this branch.


~/mywork/renderdoc/build/bin$ gdb --args ./renderdoccmd capture -d -c ./fbo
GNU gdb (Ubuntu 7.7.1-0ubuntu5~14.04.2) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./renderdoccmd...done.
(gdb) b posix_libentry.cpp:34
No source file named posix_libentry.cpp.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (posix_libentry.cpp:34) pending.
(gdb) r
Starting program: /home/mywork/renderdoc/build/bin/renderdoccmd capture -d -c ./fbo
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Breakpoint 1, library_loaded () at /home/mywork/renderdoc/renderdoc/os/posix/posix_libentry.cpp:34
34        if(LibraryHooks::Detect("renderdoc__replay__marker"))
(gdb) n
36          RDCDEBUG("Not creating hooks - in replay app");
(gdb)

**Comment by baldurk** [comment]
You're still debugging the `renderdoccmd` process so hitting that statement is entirely expected. Unless the log print indicates that target process is being detected as a replay application then this is not a bug.

**Comment by z00318884** [comment]
ok，I generally understand what you mean,Is the program being captured must be a program marked as replay? And how should I mark a program as replay application?

**Comment by baldurk** [comment]
No that is almost the opposite of what I'm saying. You don't need to mark or change your program at all to capture, in fact any program marked as replay such as `renderdoccmd` cannot be captured which is what I said above.

If there is an actual bug or problem you're encountering then please file a new issue about that with a clear and explicit description of what is actually happening and what is going wrong, without any assumptions or attempts to diagnose the problem. Otherwise misunderstanding or mistakes on your part as shown in this issue can waste time and obscure any actual problems.
